pkg/dockerregistry/server/repositorymiddleware.go

```diff
@@ -26,6 +26,21 @@ import (
 	"github.com/openshift/origin/pkg/image/importer"
 )
 
+const (
+	// DockerRegistryURLEnvVar is a mandatory environment variable name specifying external url of internal
+	// docker registry. All references to pushed images will be prefixed with its value.
+	DockerRegistryURLEnvVar = "DOCKER_REGISTRY_URL"
+
+	// EnforceQuotaEnvVar is an boolean environment variable that allows to turn quota enforcement on or off.
+	// By default, quota enforcement is off. Recognized values are "true", and "false".
+	EnforceQuotaEnvVar = "OPENSHIFT_ENFORCE_QUOTA"
+
+	// ProjectCacheTTLEnvVar is an environment variable specifying an eviction timeout for project quota
```

Comment:
[Middleware config option](https://github.com/docker/distribution/blob/master/configuration/configuration.go#L53) is a map of arrays. Map is [treated](https://github.com/docker/distribution/blob/master/configuration/parser.go#L167) gracefully. However, there's no a way to address anyting in anarray.

And the upstream itself [suggests to edit the config file](https://github.com/docker/distribution/issues/1109) instead of overriding complex options with env vars.